src/entities.py

```diff
@@ -95,193 +95,241 @@ def __hash__(self) -> int:
-        "Senf": "Sf",
-    }
-
-    mediziner_ingredient_lookup = {
-        "1": "1",
-        "2": "2",
-        "3": "3",
-        "4": "4",
-        "5": "5",
-        "6": "6",
-        "7": "7",
-        "8": "8",
-        "9": "9",
-        "A": "99",
-        "B": "Gl",
-        "C": "Kr",
-        "E": "Fi",
-        "F": "Fi",
-        "G": "G",
-        "H": "En",
-        "K": "K",
-        "L": "L",
-        "M": "So",
-        "N": "Mi",
-        "O": "Sc",
-        "P": "Sl",
-        "R": "R",
-        "S": "S",
-        "T": "Sf",
-        "U": "Se",
-        "V": "Sw",
-        "W": "W",
-        "X": "Lu",
-        "Y": "Ei",
-        "Z": "Wt",
-    }
-
-    def __init__(self, location: str):
-        self.location = location
-        self.ingredient_set = set()
-
-    def _values_lookup(self, values: Sequence[str], lookup: Optional[Dict[str, str]]) -> None:
-        """
-        Normalizes ingredients to the self.ingredient_lookup codes.
-
-        Args:
-            values: A sequence of ingredients codes.
-            lookup: If needed, a mapping from a canteen specific ingredients codes to the self.ingredient_lookup codes.
-        """
-        for value in values:
-            # ignore empty values
-            if not value or value.isspace():
-                continue
-            if (not lookup and value not in self.ingredient_lookup) or (lookup and value not in lookup):
-
-                # sometimes the ‘,’ is missing between the ingredients (especially with IPP) and we try to split again
-                # with capital letters.
-                split_values: List[Any] = re.findall(r"[a-züöäA-ZÜÖÄ][^A-ZÜÖÄ]*", value)
-                if split_values:
-                    self._values_lookup(split_values, lookup)
-                    continue
-                else:
-                    print("Unknown ingredient for " + self.location + " found: " + str(value))
-                    continue
-
-            if lookup:
-                self.ingredient_set.add(lookup[value])
-            else:
-                self.ingredient_set.add(value)
-
-    def parse_ingredients(self, values: str) -> None:
-        """
-        Parse and creates a normalized list of ingredients.
-
-        Args:
-            values: String with comma separated ingredients codes.
-        """
-        values = values.strip()
-        split_values: List[str] = values.split(",")
-        # check for special parser/ingredient translation required
-        if self.location == "fmi-bistro":
-            self._values_lookup(split_values, self.fmi_ingredient_lookup)
-        elif self.location == "mediziner-mensa":
-            self._values_lookup(split_values, self.mediziner_ingredient_lookup)
-        # default to the "Studentenwerk" ingredients
-        # "ipp-bistro" also uses the "Studentenwerk" ingredients since all
-        # dishes contain the same ingredients
-        else:
-            self._values_lookup(split_values, None)
+class Site:
```

Comment:
I renamed it to Location. Since previously a canteen was sometimes refered to as canteen and sometimes as location, I now changed it so that a canteen is consistently refered to as canteen.